src/tests/lib/repositories/impl/inventory_ingredient_repository_impl_test.py

```diff
@@ -0,0 +1,119 @@
+
+        inventory_ingredient_repository.add(inventory_ingredients_to_insert[0])
+        inventory_ingredient_repository.add(inventory_ingredients_to_insert[1])
+        inventory_ingredient_repository.add(inventory_ingredients_to_insert[2])
+        inventory_ingredient_repository.add(inventory_ingredients_to_insert[3])
+        inventory_ingredient_repository.add(inventory_ingredients_to_insert[4])
+
+        inventory_ingredients = inventory_ingredient_repository.get_all()
+
+        self.assertEqual(
+            inventory_ingredients,
+            [
+                inventory_ingredients_to_insert[0],
+                inventory_ingredients_to_insert[1],
+                inventory_ingredients_to_insert[2],
+                inventory_ingredients_to_insert[3],
+                inventory_ingredients_to_insert[4],
+            ],
+        )
+
+    def test_get_all_inventory_ingredients_empty_successfully(self):
+        inventory_ingredient_repository = InventoryIngredientRepositoryImpl()
+
+        inventory_ingredients = inventory_ingredient_repository.get_all()
+
+        self.assertEqual(inventory_ingredients, [])
+
+    def test_delete_an_inventory_ingredient_successfully(self):
+        inventory_ingredients_to_insert = build_inventory_ingredients(count=3)
+
+        inventory_ingredient_repository = InventoryIngredientRepositoryImpl()
+
+        inventory_ingredient_repository.add(inventory_ingredients_to_insert[0])
+        inventory_ingredient_repository.add(inventory_ingredients_to_insert[1])
+        inventory_ingredient_repository.add(inventory_ingredients_to_insert[2])
+
+        inventory_ingredient_repository.delete_by_id(2)
+
+        inventory_ingredients = inventory_ingredient_repository.get_all()
+
+        self.assertEqual(
+            inventory_ingredients,
+            [inventory_ingredients_to_insert[0], inventory_ingredients_to_insert[2]],
+        )
+
+    def test_delete_throws_key_error_when_there_are_no_inventory_ingredients(self):
+        inventory_ingredient_repository = InventoryIngredientRepositoryImpl()
+
+        self.assertRaises(KeyError, inventory_ingredient_repository.delete_by_id, 2)
+
+    def test_update_inventory_ingredient_successfully(self):
+        inventory_ingredients_to_insert = build_inventory_ingredients(count=2)
+
+        inventory_ingredient_repository = InventoryIngredientRepositoryImpl()
+
+        inventory_ingredient_repository.add(inventory_ingredients_to_insert[0])
+        inventory_ingredient_repository.add(inventory_ingredients_to_insert[1])
+
+        inventory_ingredient_to_update = build_inventory_ingredient(
+            ingredient_quantity=2
+        )
+
+        inventory_ingredient_repository.update_by_id(2, inventory_ingredient_to_update)
+        updated_inventory_ingredient = inventory_ingredient_repository.get_by_id(2)
+        inventory_ingredients = inventory_ingredient_repository.get_all()
+
+        self.assertEqual(len(inventory_ingredients), 2)
```

Comment:
This assert equals seems useless

Reply:
why not, this assert is checking that you are not adding another inventory_ingredidient to the inventory_ingredients list, just
updating one inventory_ingredient that is already inside the list